pkgs/test_core/lib/src/runner/reporter/github.dart

```diff
@@ -113,6 +114,9 @@ class GithubReporter implements Reporter {
       }
     }
 
+    // TODO: how to recognize (setUpAll) and (tearDownAll)?
```

Comment:
Do these show up in the other test reporters? I really don't remember them displaying in test results.

I think it's odd - if people have 3 tests defined - to display results for 5 tests. Are the results for these included in the `engine.passed` set of tests? Again, I think it would be odd - if they have 3 tests defined in a project - for the test run to conclude with `All 5 tests passed`.

Perhaps a setup and teardown phase can be visualized a bit differently from a regular test?

▶ test/message_extraction/failed_extraction_test.dart: foo.setUpAll()
▶ test/message_extraction/failed_extraction_test.dart: foo.tearDownAll()

Reply:
> Do these show up in the other test reporters? I really don't remember them displaying in test results.

They do show up today yes (you would usually only see them if they fail, like all other tests in the compact reporter). On github (expanded reporter) you definitely see them though (something like `00:00 +0: (setUpAll)`).

> I think it's odd - if people have 3 tests defined - to display results for 5 tests. 

Yes, it is a bit odd and the current reporters _don't_ count these in the total number of "successes". It gets weirder if they fail, they do count as a failure, and then no other tests run at all so it reports one test failing regardless of how many tests are in the suite (the others disappear basically).

> Perhaps a setup and teardown phase can be visualized a bit differently from a regular test?

I am fine with them being visualized differently, to some extent this already happens (the name is `(setUpAll)` in the example runs here). Not sure if it would also show the group name, but that would be good.

Reply:
From some experimentation, setUp() and tearDown() don't cause new tests to be created; setUpAll() and tearDownAll() do.

They're comparatively rare, so I think what we do here isn't critical. I'll likely update this to display them using groups, and make sure the group doesn't look like a normal, user-authored test.

Reply:
> From some experimentation, setUp() and tearDown() don't cause new tests to be created; setUpAll() and tearDownAll() do.

Ya, because these run _per test_ they are considered just a part of the test. The thing with `setUpAll` is it runs once, before all the tests, and so you can't link it to any one of them. I guess you could link it to _all_ of them though? But I think that just ends up being noisy, would duplicating the same error in each test.

Reply:
Gotcha - thanks for the background, makes sense.